plugins/channel-safely/include/channel-manager.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include <PluginManager.h>
+#include <modules/World.h>
+#include <modules/Maps.h>
+#include <modules/Job.h>
+#include <df/map_block.h>
+#include "channel-groups.h"
+#include "plugin.h"
+
+using namespace DFHack;
+
+// Uses GroupData to detect an unsafe work environment
+class ChannelManager {
+private:
+    ChannelJobs jobs;
+    ChannelManager()= default;
+protected:
+public:
+    ChannelGroups groups = ChannelGroups(jobs);
+
+    static ChannelManager& Get(){
+        static ChannelManager instance;
+        return instance;
+    }
+
+    void build_groups() { groups.scan(); debug(); }
+    void destroy_groups() { groups.clear(); debug(); }
+    void manage_groups();
+    void manage_group(const df::coord &map_pos, bool set_marker_mode = false, bool marker_mode = false);
+    void manage_group(const Group &group, bool set_marker_mode = false, bool marker_mode = false);
+    bool manage_one(const Group &group, const df::coord &map_pos, bool set_marker_mode = false, bool marker_mode = false);
+    void mark_done(const df::coord &map_pos);
+    bool exists(const df::coord &map_pos) const { return groups.count(map_pos); }
+    void debug() {
+        DEBUG(groups).print(" DEBUGGING GROUPS:\n");
+        if (config.debug) {
```

Comment:
do we need an additional state variable to track this or can we just use the existing `debugfilter`-managed categories?

Reply:
`config.debug` it is necessary for that exact use iirc (enables the code if any debugging is going on)

Reply:
that's what I'm confused about -- doesn't `debugfilter` already track the verbosity level for all your categories? why do you need another tracker here?

Reply:
It was because I wanted it to get to this function if any debugging was taking place